### PR TITLE
Merge 4.0 into master

### PR DESCRIPTION
**Pull request by pchasle**

### CODE REVIEW

**Comment by pchasle** [comment]
@ahocquard I see that you have added php 7.1 support, does this mean that we should update the `README.md` "~_php >= 7.2_~" to "_php >= 7.1_" and the `docker-compose.yml` ?

**Review by fitn — APPROVED** [review]
I let Alex analyzes this but it seems weird to remove some code between 4.x to master, right?

**Review by ahocquard — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 language: php
 
 php:
+    - 7.1
```
Comment (ahocquard) [inline .travis.yml:4]: missing composer in the pull-up :thinking:

**Comment by ahocquard** [comment]
@fitn It was a  bug: the endpoint does not exist.

**Comment by ahocquard** [comment]
@pchasle I would not update docker. I would keep php 7.2 and let the CI do the check for PHP 7.1.

File: .travis.yml (hunk 1 above)
Comment (pchasle) [reply .travis.yml:4]: Nothing changed in the `composer.json` but I need to pull-up the `akeneo/api-php-client` too

Comment (pchasle) [reply .travis.yml:4]: I will do a composer update once the CE version of the php-client is merged.
With probably some fixes for phpunit 8

**Comment by pchasle** [comment]
CI is green.
The master branch of `api-php-client` was not merged yet when the CI triggered. So I relaunched it manually after and it passed.

**Pull request merged**